docs/data/sdk-plugins.md

```diff
@@ -403,4 +403,95 @@ Use a Destination Plugin to send events to a third-party APIs
+
+                constructor(containerId) {
+                  this.containerId = containerId
+                }
+
+                injectGtmScript(containerId) {
+                  if (!window.dataLayer) {
+                    window.dataLayer = window.dataLayer || []
+                    window.dataLayer.push({
+                      "gtm.start": new Date().getTime(),
+                      event: "gtm.js"
+                    })
+                    const head = document.getElementsByTagName("head")[0],
+                      script = document.createElement("script"),
+                      dataLayer = "datalayer" != "dataLayer" ? "&l=" + "datalayer" : ""
+                    script.async = true
+                    script.src =
+                      "https://www.googletagmanager.com/gtm.js?id=" + containerId + dataLayer
+                    head.insertBefore(script, head.firstChild)
+                  }
+                }
+
+                async setup() {
+                  this.injectGtmScript(this.containerId)
+                }
+
+                async execute(event) {
+                  this.injectGtmScript(this.containerId)
+                  window.dataLayer.push(event)
+
+                  return {
+                    code: 0,
+                    event: event,
+                    message: "Event pushed onto GTM Data Layer"
+                  }
+                }
+              }
+            ```
+        
+        === "TypeScript"
+        
+            ```ts
+            import { DestinationPlugin, Event, PluginType, Result } from '@amplitude/analytics-types';
+            export class GTMPlugin implements DestinationPlugin {
+              name = 'google-tag-manager';
+              type = PluginType.DESTINATION as const;
+              containerId: string;
+
+              constructor(containerId: string) {
+                this.containerId = containerId;
+              }
+
+              injectGtmScript(containerId: string) {
+                if (!window.dataLayer) {
+                  window.dataLayer = window.dataLayer || [];
+                  window.dataLayer.push({ 'gtm.start': new Date().getTime(), event: 'gtm.js' });
+                  const head = document.getElementsByTagName('head')[0],
+                    script = document.createElement('script'),
+                    dataLayer = 'datalayer' != 'dataLayer' ? '&l=' + 'datalayer' : '';
+                  script.async = true;
+                  script.src = 'https://www.googletagmanager.com/gtm.js?id=' + containerId + dataLayer;
+                  head.insertBefore(script, head.firstChild);
+                }
+              }
+
+              async setup(): Promise<void> {
+                this.injectGtmScript(this.containerId);
+              }
+
+              async execute(event: Event): Promise<Result> {
+                this.injectGtmScript(this.containerId);
+                window.dataLayer.push(event);
+
+                return {
+                  code: 0,
```

Comment:
Maybe status code 200?

```suggestion
                  code: 200,
```

Reply:
+1, otherwise there will be `Amplitude Logger [Error]: Event pushed onto GTM Data Layer`